packages/react-router/tests/redirect.test.tsx

```diff
@@ -252,6 +252,84 @@ describe('redirect', () => {
       expect(await screen.findByText('Final')).toBeInTheDocument()
       expect(window.location.pathname).toBe('/final')
     })
+
+    test('when `redirect` is thrown in route component', async () => {
```

Comment:
I think we should test the back button behaviour here since it'd likely be one of the few buttons that'd be front of mind for the end-user to consider immediately hitting next.